src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java

```diff
@@ -17,27 +18,43 @@ public OrderPage openOrderPage() {
     }
 
     public BasketPage increaseProductCount(int numberOfProduct) {
-        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[2])[%d]", numberOfProduct)).click();
+        var productSum = requireNonNull($x(format("(//*[@class='cart-product__price'])[%d]", numberOfProduct))
+                .text());   //added for waiting the information in the basket window to be updated
+        $x(format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[2])[%d]",
+                numberOfProduct)).click();
+        $x(format("(//*[@class='cart-product__price'])[%d]", numberOfProduct))
+                .shouldNotHave(text(productSum));   //added for waiting the information in the basket window to be updated
         return this;
     }
 
     public BasketPage decreaseProductCount(int numberOfProduct) {
-        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[1])[%d]", numberOfProduct)).click();
+        var productSum = requireNonNull($x(format("(//*[@class='cart-product__price'])[%d]", numberOfProduct))
+                .text());  //added for waiting the information in the basket window to be updated
+
+        if (!$x(format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[1])[%d]",
+                numberOfProduct)).isEnabled()) { return this; }
```

Comment:
this will exit method without doing anything

Reply:
exception added